### PR TITLE
1.1 Add Iio makefile

### PR DESCRIPTION
**Pull request by mchindri**
Last commit is relevant.
On top of https://github.com/analogdevicesinc/no-OS/pull/657

### CODE REVIEW

**Comment by mchindri** [comment]
v2:
 - Check for .git instead of README.md. Solves https://github.com/analogdevicesinc/no-OS/pull/657#discussion_r444234264
 - Remove lines used for testing only. Solves https://github.com/analogdevicesinc/no-OS/pull/657#discussion_r443365326

**Comment by mchindri** [comment]
v3:
 - Replace $(OBJS) with $^ . Solves this: https://github.com/analogdevicesinc/no-OS/pull/657#discussion_r444239694

**Comment by mchindri** [comment]
v4:
 - Rebase onto master

File: libraries/iio/Makefile [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+SRCS = iio.c $(TINYIIOD_DIR)/parser.c $(TINYIIOD_DIR)/tinyiiod.c
+
+OBJS = $(SRCS:.c=.o)
+
+CFLAGS += $(IIO_DEFS) \
+	  -I"$(TINYIIOD_DIR)"
+
+.PHONEY = all clean
+
+#Init tiniiod submodule if not initialized
```
Comment (buha) [inline libraries/iio/Makefile:39]: typo in comment

File: libraries/iio/Makefile [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+IIO_DEFS = -D TINYIIOD_VERSION_MAJOR=$(TINYIIOD_VERSION_MAJOR)		\
+	   -D TINYIIOD_VERSION_MINOR=$(TINYIIOD_VERSION_MINOR)		\
+	   -D TINYIIOD_VERSION_GIT=0x$(shell git rev-parse --short HEAD)\
+	   -D IIOD_BUFFER_SIZE=$(BUFF_SIZE)
+
+ifeq (true,$(strip $(STD_TYPES)))
+IIO_DEFS += -D_USE_STD_INT_TYPES
+endif
+
+IIO_DIR		= $(realpath .)
```
Comment (buha) [inline libraries/iio/Makefile:25]: I'm not very familiar with makefile syntax but i was wondering if `.` could point to different directory than `libraries/iio`. I tend to think that it's not possible so this should be ok.

**Comment by buha** [comment]
Fix that typo in comment, the rest looks good.

File: libraries/iio/Makefile (hunk 2 above)
Comment (mchindri) [reply libraries/iio/Makefile:25]: So, when "make -C libraries/iio" is run from no-os makefile directory will change to libraries/iio -> run make -> go back to last directory.
So yes, "." will point to libraries/iio

File: libraries/iio/Makefile (hunk 1 above)
Comment (mchindri) [reply libraries/iio/Makefile]: done

**Comment by mchindri** [comment]
v5: 
 - Fix comment typo: tiniiod -> libtinyiiod

**Comment by mchindri** [comment]
Will merge today if no more comments

**Pull request merged**